### ISSUE TITLE
WinSock2.h conflicting with some header

### ISSUE THREAD

**Issue by FabrizioDys** [MEPP-team/MEPP2#28]
Good morning.
I am writing you because I am currently trying to include WinSock2 to estabilish an HTTP connection to retrieve a mesh from the internet, the thing is when I compile my solution the library conflicts with the already present winsock members.. So, looking in the internet, I found out that windows.h automatically includes winsock.h . I searched in the full solution if there were any traces of "windows.h" but I didn't find any. Can you help me solve this?

**Comment by elombardi2** [comment]
I agree it's weird that `windows.h` doesn't appear anywhere. I suppose it is included by an external header.

Try to define [WIN32_LEAN_AND_MEAN](https://stackoverflow.com/questions/5971332/redefinition-errors-in-winsock2-h) at the CMake level so that it's defined before `windows.h` is included: add the line `add_definitions(-DWIN32_LEAN_AND_MEAN)` [here](https://github.com/MEPP-team/MEPP2/blob/24cd2a293fbc00cf53e8f96eea5450ee88307c8a/Cmake/CompilerOptions.cmake#L51).

**Comment by mtola** [comment]
Found at first search here (but it is undoubtedly necessarily present in other dependencies...)
https://github.com/MEPP-team/MEPP2/blob/dd04373dd7eb9e0508dc621e53f8ee4cba41d8ac/External/CGAL-4.14/CGAL/config.h#L46

Yes, winsock2.h must be included before windows.h, see here:
https://stackoverflow.com/questions/21399650/cannot-include-both-files-winsock2-windows-h

---> But for me you are on a very very bad way, I already played with winsock 25 years back under Win95, during my studies... For me, this is rather old and deprecated solution : method of another time.
Moreover, this solution is not portable, it will only work on Windows.
You must use a solution that will work on Windows, Linux and Mac.

I think it is necessary to take (really) the time to learn CMake, Boost, LibCurl or another library with a portable solution, in a separate example...

Good easter we.

**Comment by mtola** [comment]
...
And I think also you will have problems with 32/64 bits architecture with that...
MEPP2 is only 64 bits.

**Comment by FabrizioDys** [comment]
Thank you for the response. We tried using libCurl but there were so many issues we decided to let it go. For Boost there were the same problems that I am explaining here ( the WinSock2 conflict with windows.h header files ), so I think I will continue on the other components of Boost that are already present in local_vs2015_64 and try to follow the solution @elombardi2 gave me. Thank you for the suggestion, neither am I a fan of using sockets in 2020 if not strictly necessary
